tooling/module-domain/src/main/java/org/eclipse/dataspaceconnector/tooling/module/domain/ModuleType.java

```diff
@@ -0,0 +1,43 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.tooling.module.domain;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+
+/**
+ * Enumerates {@link EdcModule} types. SPI modules define extension points and extension modules contribute capabilities to a runtime.
+ */
+public enum ModuleType {
+    SPI("spi"),
+    EXTENSION("extension");
```

Comment:
What about core modules? Will they also be classified as extension?

Reply:
Yes. They may have a subclassification but from the runtime's standpoint, they are all extensions.